app/src/main/java/com/zeapo/pwdstore/LaunchActivity.kt

```diff
@@ -40,18 +50,39 @@ class LaunchActivity : AppCompatActivity() {
     }
 
     private fun startTargetActivity(noAuth: Boolean) {
-        val intentToStart = if (intent.action == ACTION_DECRYPT_PASS)
-            Intent(this, DecryptActivity::class.java).apply {
+        if (intent.action == ACTION_DECRYPT_PASS) {
+            val intent = Intent(this, DecryptActivity::class.java).apply {
                 putExtra("NAME", intent.getStringExtra("NAME"))
                 putExtra("FILE_PATH", intent.getStringExtra("FILE_PATH"))
                 putExtra("REPO_PATH", intent.getStringExtra("REPO_PATH"))
                 putExtra("LAST_CHANGED_TIMESTAMP", intent.getLongExtra("LAST_CHANGED_TIMESTAMP", 0L))
+                startPasswordStoreActivity(intent, noAuth)
             }
-        else
-            Intent(this, PasswordStore::class.java)
-        startActivity(intentToStart)
+        } else {
+            val intent = Intent(this, PasswordStore::class.java)
+            if (isInitialized && !GitSettings.url.isNullOrEmpty() && sharedPrefs.getBoolean(PreferenceKeys.SYNC_ON_LAUNCH, false)) {
+                findViewById<ProgressBar>(R.id.progress_bar).isVisible = true
+                findViewById<ProgressBar>(R.id.sync).isVisible = true
+                runGitOperation(intent)
+            } else {
+                startPasswordStoreActivity(intent, noAuth)
+            }
+        }
+    }
+
+    private fun startPasswordStoreActivity(intent: Intent, noAuth: Boolean) {
+        Handler().postDelayed({
+            startActivity(intent)
+            finish()
+        }, 1000L)
```

Comment:
```suggestion
        }, if (noAuth) 1000L else 0L)
```
I still much prefer this